src/DocumentWatcher.ts

```diff
@@ -172,6 +173,22 @@ function applyEditorConfigToTextEditor(
 	/* tslint:disable:no-any */
 	textEditor.options = <any> newOptions;
 	/* tslint:enable */
+
+	applyEndOfLineSetting();
+
+	function applyEndOfLineSetting() {
+		for (const [regex, eol] of new Map([
+			[/^lf$/i, EndOfLine.LF],
+			[/^crlf$/i, EndOfLine.CRLF]
+		])) {
+			if (regex.test(editorconfig.end_of_line)) {
+				textEditor.edit(edit => {
+					edit.setEndOfLine(eol);
+				});
+				return;
+			}
+		}
```

Comment:
With editorconfig, simply opening a text file shouldn't make a file dirty. That's my point. We've had this discussion before on the team.